src/compute-client/src/controller/replica.rs

```diff
@@ -179,36 +183,59 @@ where
     /// The connection is retried forever (with backoff) and this method returns only after
     /// a connection was successfully established.
     async fn connect(&self) -> Client<T> {
-        let try_connect = |retry: RetryState| {
-            let addrs = &self.config.location.ctl_addrs;
-            let dests = addrs
-                .iter()
-                .map(|addr| (addr.clone(), self.metrics.clone()))
-                .collect();
+        let try_connect = async |retry: RetryState| {
             let version = self.build_info.semver_version();
             let client_params = &self.config.grpc_client;
```

Comment:
Reusing the grpc params for CTP connections here. Once the gRPC protocol is removed, we should rename them.